opentracing-spring-web/src/test/java/io/opentracing/contrib/spring/web/client/TracingAsyncRestTemplateTest.java

```diff
@@ -40,4 +50,42 @@ public AsyncRestTemplate template() {
 
         mockServer = MockRestServiceServer.bindTo(client.template()).build();
     }
+
+    @Test
+    public void testMultipleRequests() throws InterruptedException {
```

Comment:
I don't think this test is exercising the tricky parts of an async setup :-) Ideally, you'd ensure that data didn't leak from one thread to another. 

For instance, you could have 5 calls to each of those:

`http://localhost:8080/foo/span1`, `http://localhost:8080/foo/span2`, `http://localhost:8080/foo/span3`. The parent span for `/foo/span1` is `span1` and you'd check afterwards if the parent span you got on `span1` was seen as `span1`. This way, you are testing that your data is protected even in the case of concurrent async requests.